Sources/Workspace/DefaultPluginScriptRunner.swift

```diff
@@ -353,47 +331,43 @@ public struct DefaultPluginScriptRunner: PluginScriptRunner, Cancellable {
         }
         
         // Now invoke the compiler asynchronously.
-        AsyncProcess.popen(arguments: commandLine, environment: environment, queue: callbackQueue) {
-            // We are now on our caller's requested callback queue, so we just call the completion handler directly.
-            dispatchPrecondition(condition: .onQueue(callbackQueue))
-            completion($0.tryMap { process in
-                // Emit the compiler output as observable info.
-                let compilerOutput = ((try? process.utf8Output()) ?? "") + ((try? process.utf8stderrOutput()) ?? "")
-                if !compilerOutput.isEmpty {
-                    observabilityScope.emit(info: compilerOutput)
-                }
+        let process = try await AsyncProcess.popen(arguments: commandLine, environment: environment)
 
-                // Save the persisted compilation state for possible reuse next time.
-                let compilationState = PersistedCompilationState(
-                    commandLine: commandLine,
-                    environment: toolchain.swiftCompilerEnvironment.cachable,
-                    inputHash: compilerInputHash,
-                    output: compilerOutput,
-                    result: .init(process.exitStatus))
-                do {
-                    try JSONEncoder.makeWithDefaults().encode(path: stateFilePath, fileSystem: self.fileSystem, compilationState)
-                }
-                catch {
-                    // We couldn't write out the `.state` file. We warn about it but proceed.
-                    observabilityScope.emit(debug: "Couldn't save plugin compilation state", underlyingError: error)
-                }
+        // Emit the compiler output as observable info.
+        let compilerOutput = ((try? process.utf8Output()) ?? "") + ((try? process.utf8stderrOutput()) ?? "")
+        if !compilerOutput.isEmpty {
+            observabilityScope.emit(info: compilerOutput)
+        }
 
-                // Construct a PluginCompilationResult for both the successful and unsuccessful cases (to convey diagnostics, etc).
-                let result = PluginCompilationResult(
-                    succeeded: compilationState.succeeded,
-                    commandLine: commandLine,
-                    executableFile: execFilePath,
-                    diagnosticsFile: diagFilePath,
-                    compilerOutput: compilerOutput,
-                    cached: false)
-
-                // Tell the delegate that we're done compiling the plugin, passing it the result.
-                delegate.didCompilePlugin(result: result)
-                
-                // Also return the result to the caller.
-                return result
-            })
+        // Save the persisted compilation state for possible reuse next time.
+        let compilationState2 = PersistedCompilationState(
```

Comment:
Looks like a typo

```suggestion
        let compilationState = PersistedCompilationState(
```

Reply:
it is not there are now two in this scope